src/app/profile-edit/profile-edit.page.ts

```diff
@@ -0,0 +1,139 @@
+            if (control.dynamicEntity) {
+              this.getOptionsData(control.name);
+            }
+          });
+        }
+      });
+  }
+
+  getOptionsData(entityType: string, entityId?: string) {
+    const control = this.formJson.find((control: any) => control.name === entityType);
+    if (!control) return;
+    const urlPath = control.dynamicEntity
+      ? this.urlProfilePath.entityUrl+`${entityType}`
+      : this.urlProfilePath.subEntityUrl+`/${entityId}?type=${entityType}&search=&page=1&limit=100`;
+
+    this.apiBaseService.get(urlConfig['entityListing'].listingUrl + urlPath)
+      .pipe(
+        catchError(err => {
+          this.toastService.presentToast(err?.error?.message);
+          return of(null);
+        })
+      )
+      .subscribe((res:any) => {
+        if (res?.status === 200) {
+          const result = control.dynamicEntity ? res?.result : res?.result?.data;
+          if(result){
+            const options = result.map((entity: any) => ({
+              label: entity.name,
+              value: entity._id,
+            }));
+            this.updateFormOptions(entityType, options);
+          }
+        } else {
+          this.toastService.presentToast(res.message);
+        }
+      });
+  }
+
+  updateFormOptions(entityType: string, options: any) {
+    const control = this.formJson.find((control: any) => control.name === entityType);
+    if (control) {
+      control.options = options;
+    }
+  }
+
+  onOptionChange(event: any) {
+    const { event: selectedEvent, control } = event;
+    const selectedValue = selectedEvent?.value;
+    this.updateFormValue(control.name, selectedValue?.value);
+    this.formJson.forEach((formControl: any) => {
+      if (formControl.dependsOn === control.name) {
+        this.resetFormControl(formControl.name);
+        this.getOptionsData(formControl.name, selectedValue?.value);
+      }
+    });
+  }
+
+  getNextEntityType(currentEntityType: string): string | null {
+    const nextControl = this.formJson.find((ctrl: any) => ctrl.dependsOn === currentEntityType);
+    return nextControl ? nextControl.name : null;
+  }
+
+  resetFormControl(controlName: string) {
+    const control = this.formLib?.myForm.get(controlName);
+    if (control) {
+      control.patchValue('');
+      control.markAsPristine();
+      control.markAsUntouched();
+    }
+  }
+
+  updateFormValue(controlName: string, value: any) {
+    const control = this.formJson.find((formControl: any) => formControl.name === controlName);
+    if (control) {
+      control.value = value;
+    }
+  }
+
+  updateProfile() {
+    if (this.formLib?.myForm.valid) {
+      this.apiBaseService.patch(this.urlProfilePath.updateUrl, this.formLib.myForm.value)
+        .pipe(
+          catchError(err => {
+            this.toastService.presentToast(err?.error?.message);
+            return of(null);
+          })
+        )
+        .subscribe((res:any) => {
+          if (res?.status === 200) {
```

Comment:
why is this block empty for successful response? shouldn't use see success message if the profile is updated?